libs/remix-ui/static-analyser/src/lib/remix-ui-static-analyser.tsx

```diff
@@ -151,28 +202,74 @@ export const RemixUiStaticAnalyser = (props: RemixUiStaticAnalyserProps) => {
               warningMessage.push({ msg, options, hasWarning: true, warningModuleName: moduleName })
             })
           })
-          const resultArray = []
-          warningMessage.map(x => {
-            resultArray.push(x)
-          })
-          function groupBy (objectArray, property) {
-            return objectArray.reduce((acc, obj) => {
-              const key = obj[property]
-              if (!acc[key]) {
-                acc[key] = []
-              }
-              // Add object to list for given key's value
-              acc[key].push(obj)
-              return acc
-            }, {})
+          // Slither Analysis
+          if (slitherEnabled) {
+            props.analysisModule.call('solidity-logic', 'getCompilerState').then((compilerState) => {
```

Comment:
Major analysis and computations like this should be abstracted from the ui component.

Reply:
We can plan for a refactor at a later date.

Reply:
Yes, refactoring will be required.